lib/connectors/memory.js

```diff
@@ -461,6 +460,13 @@ function applyFilter(filter) {
     if (typeof value === 'string' && (example instanceof RegExp)) {
       return value.match(example);
     }
+
+    if (example.regexp && typeof value !== 'undefined') {
```

Comment:
This logic should be refactored to dao.js to normalize the operand of `regexp` into an instance of `RegExp`